Update so that it actually builds.

kindaro opened the pull request:
* Advance stack snapshot version.
* Fix the breakage caused by changes in Cabal.
* Bump minor version.

This still leaves many warnings to be addressed. Only erroneous faults
were fixed due to urgency.

kindaro committed:
Update so that it actually builds.

* Advance stack snapshot version.
* Fix the breakage caused by changes in Cabal.
* Bump minor version.

This still leaves many warnings to be addressed. Only erroneous faults
were fixed due to urgency.

simonmichael commented:
Ping. Thanks for hpack-convert, could this fix be released ?

hiratara commented:
I made a patch for building `hpack-convert` with LTS12.1 .

8695b5666c8ee66785c43074b9ae8b3a5d445192

borkdude commented:
Running into this as well.

vyorkin commented on stack.yaml:
```diff
@@ -1,5 +1,4 @@
-flags: {}
 packages:
```
pls correct me if I'm wrong, but I believe it should be:
```
packages: []
```
otherwise I got error:
```
Error while parsing $.library.dependencies -
expected Array, Object, or String, encountered Null
```

```
$ hpack --version
hpack version 0.28.2
```

expipiplus1 commented:
This works well for me, can it be merged?

wbadart committed:
Got it to compile with stack LTS-15.5

Most of the changes are due to the Pretty class that was added, it
seems, in v2.2.0.1-tgt. There were some slightly trickier cases around
some data types that don't seem to be used anymore and changes to the
Package/ GenericPackge records.

Open questions:

- This doesn't pass all the tests. The main failures seem to be around
quoting in certain rendering situations and in error reporting
- Relatedly, how should the new-ish ParseResult monad be incorporated? I
opted for parseGenericPackageDescriptionMaybe because I didn't want to
infect the rest of the module with an API change, but it's unclear how
to exactly recover the old behavior.
- Is the style where it needs to be (whitespace/ indentation, ...)?

wbadart commented:
Hey all, I just opened https://github.com/kindaro/hpack-convert/pull/1. Would love to discuss if any of you are still interested in this package as of 2020.

wbadart committed:
Bump patch version, regenerate cabal file

kindaro committed:
Merge pull request #1 from wbadart/hpack-convert

Got it to compile with stack LTS-15.5
